### PR TITLE
AppId availability check is returning a 500 in admin console.

### PR DESCRIPTION
**Pull request by jonashendrickx**
`GET /admin/apps/{appId}/available` returns a `409` when checking whether an `AppId` is available. However, since we're not modifying a resource, it should just return a `200` instead, with the expected body.

Also implemented the problem details handler, which I believe I did also before, but may have been removed. It will now properly catch any invalid payloads and throw a `PasswordlessApiException` instead of a `HttpRequestException`.

`AppId` with length less than 3 characters will return a `400` in problem details format.

Our observability tooling should no longer display unnecessary errors.

### CODE REVIEW

File: src/AdminConsole/Services/PasswordlessManagement/ProblemDetailsDelegatingHandler.cs
Comment (abergs) [inline src/AdminConsole/Services/PasswordlessManagement/ProblemDetailsDelegatingHandler.cs:1]: This is indeed implemented in the .net sdk, I suggest we try to re-use it: https://github.com/bitwarden/passwordless-dotnet/blob/854d3e5f4695318b8d8bff7edf4317f4a8097d3b/src/Passwordless/PasswordlessHttpHandler.cs#L30

**Review by abergs — COMMENTED** [review]
Assuming the issue in the comment is addressed, this looks fairly straight forward(?)

I'll let someone else double check

File: src/AdminConsole/Services/PasswordlessManagement/ProblemDetailsDelegatingHandler.cs
Comment (jonashendrickx) [reply src/AdminConsole/Services/PasswordlessManagement/ProblemDetailsDelegatingHandler.cs:1]: @abergs It is now using `PasswordlessHttpHandler`.

File: src/AdminConsole/Endpoints/ApplicationEndpoints.cs [hunk 1]
```diff
@@ -13,8 +13,18 @@ public static IEndpointRouteBuilder MapApplicationEndpoints(this IEndpointRouteB
...
+                var result = await client.IsApplicationIdAvailableAsync(request);
+                return Ok(result);
+            }
+            catch (PasswordlessApiException e)
+            {
+                return Json(
+                    e.Details,
+                    contentType: "application/problem+json",
+                    statusCode: e.Details.Status);
+            }
```
Comment (Tyrrrz) [inline src/AdminConsole/Endpoints/ApplicationEndpoints.cs:27]: In API, we do this inside a middleware for all endpoints. Maybe we should do that here too?

File: src/Api/Endpoints/Apps.cs [hunk 2]
```diff
@@ -127,6 +116,12 @@ public static void MapAccountEndpoints(this WebApplication app)
             .RequireCors("default");
     }
 
+    public static async Task<IResult> IsAppIdAvailableAsync([AsParameters] GetAppIdAvailabilityRequest payload, ISharedManagementService accountService)
```
Comment (Tyrrrz) [inline src/Api/Endpoints/Apps.cs:119]: Personally prefer the method to be inlined as it was before

File: src/AdminConsole/Endpoints/ApplicationEndpoints.cs (hunk 1 above)
Comment (abergs) [reply src/AdminConsole/Endpoints/ApplicationEndpoints.cs:27]: @Tyrrrz Not sure I'm thinking correctly, but: Adminconsole doesn't return Json in most cases (this being a bit special endpoint). I think(?) we have a generic exception page that gets rendered for uncaught exceptions

Comment (Tyrrrz) [reply src/AdminConsole/Endpoints/ApplicationEndpoints.cs:27]: Ah I see. Makes sense.

File: src/Api/Endpoints/Apps.cs (hunk 2 above)
Comment (jonashendrickx) [reply src/Api/Endpoints/Apps.cs:119]: No

**Comment by abergs** [comment]
shape: I'm OK with these changes. I'll delegate other review/approval to @Tyrrrz or @jrmccannon

**Review by jrmccannon — APPROVED** [review]

**Review by Tyrrrz — APPROVED** [review]

**Pull request merged**